miio/g1vacuum.py

```diff
@@ -0,0 +1,312 @@
+    "clean_time": {"siid": 18, "piid": 5},
+    "total_clean_count": {"siid": 9, "piid": 5},
+    #   "total_clean_area": {"siid": 9, "piid": 3},
+    #  "dnd_enabled": {"siid": 12, "piid": 2},
+    #  "audio_volume": {"siid": 4, "piid": 2},
+    #  "direction_key": {"siid": 8, "piid": 1}
+}
+
+class ChargeState(Enum):
+    Not_charging = 0
+    Charging = 1
+    Charging_competely = 2
+
+class Error(Enum):
+    Left_wheel_error = 1
+    Right_wheel_error = 2
+    Cliff_error = 3
+    Low_battery_error = 4
+    Bump_error = 5
+    Main_brush_error = 6
+    Side_brush_error = 7
+    Fan_motor_error = 8
+    Dustbin_error = 9
+    Charging_error = 10
+    No_water_error = 11
+    Everything_is_ok = 0
+    Pick_up_error = 12
+
+class State(Enum):
+    """Vacuum Status"""
+    Idle = 1
+    Sweeping = 2
+    Paused = 3
+    Error = 4
+    Charging = 5
+    Go_Charging = 6
+
+class VacuumMode(Enum):
+    """Vacuum Mode"""
+    Global_clean = 1
+    Spot_clean = 2
+    Wiping = 3
+
+class WaterLevel(Enum):
+    """Water Flow Level"""
+    Level1 = 1
+    Level2 = 2
+    Level3 = 3
+
+class FanSpeed(Enum):
+    """Fan speeds, same as for ViomiVacuum."""
+    Mute = 0
+    Standard = 1
+    Medium = 2
+    High = 3
+
+class Languages(Enum):
+    Chinese = 0
+    English = 1
+
+class MopState(Enum):
+    Off = 0
+    On = 1
+
+class MovementDirection(Enum):
+    Left = 0
+    Right = 1
+    Forward = 2
+    Backward = 3
+    Stop = 4
+
+class G1Status:
+    """Container for status reports from the Mijia Vacuum G1."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    def battery(self) -> int:
+        """Battery Level."""
+        return self.data["battery"]
+
+    @property
+    def charge_state(self) -> ChargeState:
+        """Charging State."""
+        return ChargeState[ChargeState(self.data["charge_state"]).name]
+
+    @property
+    def error(self) -> Error:
+        """Error Message."""
+        return Error[Error(self.data["error"]).name]
+
+    @property
+    def state(self) -> State:
+        """Vacuum Status."""
+        return State[State(self.data["state"]).name]
+
+    @property
+    def fan_speed(self) -> FanSpeed:
+        """Fan Speed."""
+        return FanSpeed[FanSpeed(self.data["fan_speed"]).name]
+
+    @property
+    def operating_mode(self) -> VacuumMode:
+        """Operating Mode."""
+        return VacuumMode[VacuumMode(self.data["operating_mode"]).name]
+
+    @property
+    def mop_state(self) -> MopState:
+        """Mop State."""
+        return MopState[MopState(self.data["mop_state"]).name]
+
+    @property
+    def water_level(self) -> WaterLevel:
+        """Mop State."""
+        return WaterLevel[WaterLevel(self.data["water_level"]).name]
+
+    @property
+    def brush_life_level(self) -> int:
+        """Brush Life Level."""
+        return self.data["brush_life_level"]
+
+    @property
+    def brush_life_level2(self) -> int:
```

Comment:
What is the unit of the level? Also, please use similar naming to `ConsumableStatus` in `vacuumcontainers.py`, i.e., `side_brush` in this case, and convert the value to time delta, assuming the value indicates how long it has been since the brush has been cleaned.

Same applies for other sensory values, so I'm not going to comment them separately below.

Reply:
its just a percent level, so just an integer